Have made below changes

bhavikkothari-eaton opened the pull request:
Have made changes as below

  1. Added provision of timeout while installing libraries
  2. Added provision of path for edge-linux-test-pytest
  3. Added check for python 3.8.10

bhavikkothari-eaton committed:
Have made below changes

  1. Added provision of timeout while installing libraries
  2. Added provision of path for edge-linux-test-pytest
  3. Added check for python 3.8.10

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

markabrezovec-eaton commented on prerequisite_changes.py:
```diff
@@ -97,8 +97,13 @@ def general_changes():
+    replace("TAF/testCaseModules/keywords/setup/startup_checker.py",
+            "\"Authorization\": \"Bearer {}\"",
+            "\"X-Auth-Token= {}\"")
+    replace("TAF/testCaseModules/keywords/setup/edgex.py",
```
What does this replace do?  I assume it disables an EdgeX check which looks for services which we do not have?

bhavikkothari-eaton replied:
Yes, here we are disabling Edgex checks as we will not require this checks for manual run.

markabrezovec-eaton commented on trigger.sh:
```diff
@@ -146,14 +179,23 @@ fi
 
 result=`echo $?`
```
The placement of this 'result' is peculiar to me.  Since there are so many 'if' blocks, the last command could have been line 177, 170, or line 157.  What is the intent of this check (and subsequent validation on lines 180-186)?

If you are checking the pip install result, it needs to be moved prior to line 173.

bhavikkothari-eaton replied:
Here I am trying to ensure that "**python3 prerequisite_changes.py ${test_service} ${target}**" runs without any error. So I will move this to line 176.

markabrezovec-eaton commented:
As a prerequisite to this merge, I recommend having @rajatkumar-eaton and @SachinKumar-eaton try this on their local systems once all changes have been implemented.

bhavikkothari-eaton committed:
Made changes in trigger.sh file

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Changed path of requirements.txt

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Increase timeout limit for edgex-deployment

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Made below changes
    - Skip edge-linux-test-pytest path checking while target is not KVM in trigger.sh

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton commented:
I have tested edgex-taf execution on @SachinKumar-eaton local system. In Scahin's system python version is 3.6.8 so only observed numpy version error. except this error did not observe any error and test execution complete successfully.

rajatkumar-eaton commented on TAF/utils/scripts/manual/deploy-edgex.sh:
```diff
@@ -6,16 +6,21 @@ GLOBAL_VAR_DIR=${WORK_DIR}/TAF/config/global_variables.py
+    cd ${PYTEST_PATH}
     echo "<<<<< Installing pytest related libraries >>>>>"
-    python3 -m pip install -r ${HOME}/edge-linux-test-pytest/requirements.txt
+    timeout 2m pip3 install -r requirements.txt --use-pep517
```
@bhavikkothari-eaton is this hard coded time out if fine to have here?

@markabrezovec-eaton

bhavikkothari-eaton replied:
@rajatkumar-eaton yes, this is hard coded because any library will take approx. maximum 120 seconds to install.

rajatkumar-eaton replied:
what if we get timeout after 120s also?

bhavikkothari-eaton replied:
if timeout occurs, script will execute next line. If timeout occur while starting qemu then only main script execution will be stopped.

rajatkumar-eaton replied:
@markabrezovec-eaton please suggest here.

markabrezovec-eaton replied:
I believe the Pune team has the most issues with timeouts.  I am find with a 2m timeout if that comfortably catches the issues.  Should we error handle the return code in these cases?  We shouldn't move forward if a necessary step fails.

markabrezovec-eaton commented:
> I have tested edgex-taf execution on @SachinKumar-eaton local system. In Scahin's system python version is 3.6.8 so only observed numpy version error. except this error did not observe any error and test execution complete successfully.

Would it be possible to have @SachinKumar-eaton and @rajatkumar-eaton themselves give it a try, if they have time?  We need people who have not worked on the feature to try it themselves to get the fresh perspective.

bhavikkothari-eaton committed:
Added timeout handling provision

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Update README_Instructions.md file

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Update README_Instructions.md file

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Update README_Instructions.md file

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

bhavikkothari-eaton committed:
Made small changes in deploy_edgex.py file

Signed-off-by: BhavikKothari-eaton <BhavikKothari@eaton.com>

rajatkumar-eaton commented:
> > I have tested edgex-taf execution on @SachinKumar-eaton local system. In Scahin's system python version is 3.6.8 so only observed numpy version error. except this error did not observe any error and test execution complete successfully.
> 
> Would it be possible to have @SachinKumar-eaton and @rajatkumar-eaton themselves give it a try, if they have time? We need people who have not worked on the feature to try it themselves to get the fresh perspective.

@SachinKumar-eaton has confirmed after testing, worked fine at his end.

rajatkumar-eaton commented:
As @bhavikkothari-eaton is not able to open github account so I am merging on his behalf.
